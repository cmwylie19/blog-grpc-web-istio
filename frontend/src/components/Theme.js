import React from "react";
import {
  createMuiTheme,
  ThemeProvider,
  responsiveFontSizes,
} from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { lightGreen, grey } from "@material-ui/core/colors";

export const darkTheme = createMuiTheme({
  typography: {
    fontFamily: "Red Hat Text, sans-serif",
  },
  palette: {
    primary: { main: "#000", contrastText: "#fbfbfb" },
    secondary: { main: lightGreen[500] },
    type: "light",
  },
  props: {
    MuiButtonBase: {
      disableRipple: true,
    },
  },

  MuiButton: {
    // Name of the rule
    text: {
      // Some CSS
      color: lightGreen,
    },
  },
  MuiCssBaseline: {
    "@global": {
      "@font-face": `'Red Hat Display', sans-serif;`,
    },
  },
  MuiStepIcon: {
    root: {
      "&$active": {
        fill: lightGreen,
        "& $text": {
          fill: lightGreen,
        },
      },
    },
    text: {
      fill: lightGreen,
    },
  },
});

export const lightTheme = createMuiTheme({
  typography: {
    fontFamily: "Red Hat Text, sans-serif",
  },
  palette: {
    primary: { main: grey[700], contrastText: "#fbfbfb" },
    secondary: { main: lightGreen[500] },
    type: "light",
  },
  props: {
    MuiButtonBase: {
      disableRipple: true,
    },
  },

  MuiButton: {
    // Name of the rule
    text: {
      // Some CSS
      color: lightGreen,
    },
  },
  MuiCssBaseline: {
    "@global": {
      "@font-face": `'Red Hat Display', sans-serif;`,
    },
  },
  MuiStepIcon: {
    root: {
      "&$active": {
        fill: lightGreen,
        "& $text": {
          fill: lightGreen,
        },
      },
    },
    text: {
      fill: lightGreen,
    },
  },
});
export default function ThemeContainer({ children, theme }) {
  let responsiveTheme = responsiveFontSizes(theme);
  return (
    <ThemeProvider theme={responsiveTheme}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  );
}
