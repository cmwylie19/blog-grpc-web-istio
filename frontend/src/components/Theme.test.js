import React from "react";
import { shallow } from "enzyme";
import Theme, { darkTheme } from "./Theme";

describe("ThemeContainer", () => {
  let wrapper = shallow(
    <Theme theme={darkTheme}>
      <div id="child">child</div>
    </Theme>
  );
  it("mounts", () => {
    expect(wrapper).toBeDefined();
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
