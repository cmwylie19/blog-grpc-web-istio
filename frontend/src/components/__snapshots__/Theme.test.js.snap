// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeContainer matches snapshot 1`] = `ShallowWrapper {}`;
